### PR TITLE
#797 - Use ConfigFile where RxStorage is used for config files

### PR DESCRIPTION
**Pull request by genryxy**
Part of #797
Use `ConfigFile` where `RxStorageWrapper` is used

### CODE REVIEW

File: src/main/java/com/artipie/RepoPermissionsFromStorage.java [hunk 1]
```diff
@@ -177,11 +176,11 @@ public RepoPermissionsFromStorage(final Storage storage) {
...
-        return new RxStorageWrapper(this.storage)
-            .value(key)
-            .to(new ContentAsYaml())
-            .map(yaml -> yaml.yamlMapping(RepoPermissionsFromStorage.REPO_SECTION))
-            .to(SingleInterop.get());
+        return SingleInterop.fromFuture(
+            new ConfigFile(key).valueFrom(this.storage)
+        ).to(new ContentAsYaml())
+        .map(yaml -> yaml.yamlMapping(RepoPermissionsFromStorage.REPO_SECTION))
+        .to(SingleInterop.get());
```
Comment (olegmoz) [inline src/main/java/com/artipie/RepoPermissionsFromStorage.java:183]: @genryxy converting `new ConfigFile(...).valueFrom(...)` into `Single` and later converting this `Single` into `CompletionStage` one again seems pointless. I'd suggest to not do it and transform `CompletionStage` into result using `thenApply`, `thenCompose` etc

File: src/main/java/com/artipie/UsersFromStorageYaml.java [hunk 2]
```diff
@@ -163,14 +163,14 @@ public UsersFromStorageYaml(final Storage storage, final Key key) {
...
-            .map(YamlInput::readYamlMapping)
-            .to(SingleInterop.get());
+        return SingleInterop.fromFuture(
+            new ConfigFile(this.key).valueFrom(this.storage)
+        ).flatMap(content -> new Concatenation(content).single())
+        .map(Remaining::new)
+        .map(Remaining::bytes)
+        .map(bytes -> Yaml.createYamlInput(new String(bytes, StandardCharsets.US_ASCII)))
+        .map(YamlInput::readYamlMapping)
+        .to(SingleInterop.get());
```
Comment (olegmoz) [inline src/main/java/com/artipie/UsersFromStorageYaml.java:173]: @genryxy same concern here, I'd suggest to transform `CompletionStage` into result directly without using `Single`

**Review by olegmoz — CHANGES_REQUESTED** [review]
@genryxy thanks, please see two comments

**Comment by genryxy** [comment]
@olegmoz thanks, but I have to use converting into `Single` because `readYamlMapping` throws an `IOException` and using converting allows to catch exception. Or it would be better to catch this exception in another way? I also found code duplication in `StorageAliases#find` and replace it using `ContentAsYaml`.

**Review by olegmoz — APPROVED** [review]
@genryxy ok, understand the problem with exception. I'd rather removed `IOException` from `readYamlMapping` signature. But that is not in scope of this PR

**Pull request merged**